mantidproject/mantid
# Stop Hard Crash on JobTreeView Enter Press

**Description of work.**
Fixed the hard crash caused when pressing enter buttons in JobTreeView.
Installed an event filter to catch key presses and commit the changed data before attempting to add a new row.

**Report to:** nobody

**To test:**
1. Interfaces -> SANS -> SANS v2
2. Double click on any cell in the table and type some data
3. press either Enter (appends a new row below and edits), Shift+Enter (moves up a row and edits), or Ctrl+Enter (no functionality on SANS)
4. There should be no hard crashes, "uncaught exception" dialogue boxes, mantid warnings, or negative console output.
5. Check that Ctrl+X,V,C still have cut, paste, copy functionality

Note that some columns check data input, and will complain if you type in a random string. This is intentional. 


Fixes #23974 
Fixes #24296

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
